### ISSUE TITLE
allowInts doesn't ignore the set integers

### ISSUE THREAD

**Issue by Integralist** [mgechev/revive#743]
**Describe the bug**
I configure `allowInts: "0,1,2,3,4,5"` for the `"add-constant"` rule but still my editor reports that I should add a constant when seeing those integers.

**To Reproduce**
1. I updated revive `go get -u github.com/mgechev/revive`
2. I run it with the following flags & configuration file:

```shell
revive -config ~/revive.toml -exclude vendor/... ./...
```

```toml
enableAllRules = true

[rule.add-constant]
  severity = "error"
  maxLitCount = "3"
  allowInts = "0,1,2,3,4,5" # e.g. avoid matching `if len(...) > 0` and other simple assignments.
[rule.argument-limit]
  severity = "warning"
  arguments = [4]
[rule.atomic]
  severity = "warning"
[rule.banned-characters]
  disabled = true
[rule.bare-return]
  severity = "warning"
[rule.bool-literal-in-expr]
  severity = "warning"
[rule.cognitive-complexity]
  disabled = true
[rule.confusing-naming]
  severity = "warning"
[rule.confusing-results]
  severity = "warning"
[rule.constant-logical-expr]
  severity = "error"
[rule.context-as-argument]
  severity = "error"
[rule.context-keys-type]
  severity = "error"
[rule.cyclomatic]
  disabled = true
[rule.deep-exit]
  severity = "warning"
[rule.defer]
  severity = "warning"
[rule.early-return]
  severity = "warning"
[rule.empty-block]
  severity = "error"
[rule.empty-lines]
  severity = "warning"
[rule.error-naming]
  severity = "error"
[rule.error-return]
  severity = "error"
[rule.error-strings]
  severity = "error"
[rule.errorf]
  severity = "warning"
[rule.exported]
  severity = "error"
[rule.file-header]
  disabled = true
[rule.flag-parameter]
  severity = "warning"
[rule.function-result-limit]
  severity = "warning"
  arguments = [3]
[rule.function-length]
  severity = "warning"
  arguments = [40,0]
[rule.get-return]
  severity = "error"
[rule.identical-branches]
  severity = "error"
[rule.if-return]
  severity = "warning"
[rule.increment-decrement]
  severity = "error"
[rule.indent-error-flow]
  severity = "warning"
[rule.import-shadowing]
  severity = "warning"
[rule.line-length-limit]
  disabled = true
[rule.max-public-structs]
  disabled = true
[rule.modifies-parameter]
  severity = "warning"
[rule.modifies-value-receiver]
  severity = "warning"
[rule.nested-structs]
  severity = "warning"
[rule.optimize-operands-order]
  severity = "warning"
[rule.package-comments]
  severity = "warning"
[rule.range]
  severity = "warning"
[rule.range-val-in-closure]
  severity = "warning"
[rule.range-val-address]
  severity = "warning"
[rule.receiver-naming]
  severity = "warning"
[rule.redefines-builtin-id]
  severity = "error"
[rule.string-of-int]
  severity = "warning"
[rule.struct-tag]
  severity = "warning"
[rule.superfluous-else]
  severity = "warning"
[rule.time-equal]
  severity = "warning"
[rule.time-naming]
  severity = "warning"
[rule.var-declaration]
  severity = "warning"
[rule.var-naming]
  severity = "warning"
[rule.unconditional-recursion]
  severity = "error"
[rule.unexported-naming]
  severity = "warning"
[rule.unexported-return]
  severity = "error"
[rule.unhandled-error]
  severity = "warning"
[rule.unnecessary-stmt]
  severity = "warning"
[rule.unreachable-code]
  severity = "warning"
[rule.unused-parameter]
  severity = "warning"
[rule.unused-receiver]
  severity = "warning"
[rule.use-any]
  severity = "warning"
[rule.useless-break]
  severity = "warning"
[rule.waitgroup-by-value]
  severity = "warning"
```

**Expected behavior**
I expected lines like `if v >= 0 {` to not trigger the rule to report an error.

**Logs**
The following screenshot shows my configuration via golangci-lint, but as described above it happens when running revive directly with a similar revive.toml configuration...

<img width="2032" alt="Screenshot 2022-08-31 at 15 54 57" src="https://user-images.githubusercontent.com/180050/187712288-23cda7e2-63da-41a9-b673-a98fbefaa40a.png">

**Desktop (please complete the following information):**
 - OS: macOS Monterey 12.5
 - Tried with `1.17.13` and `1.18.5`.

**Additional context**
I'm pretty sure this was working fine initially, but then later stopped working. Don't believe any other relevant changes were made to my IDE that would affect this.

**Comment by hansgylling** [comment]
Are you expecting to install a binary of the latest version of revive when you run the command `go get -u github.com/mgechev/revive`? Because if you use Go 1.18 or later that doesn't happen. You need to use the command `go install github.com/mgechev/revive@latest` instead. This probably doesn't explain your bug, unless there has been a bugfix for the `add-constant` rule in a recent release, but it could be good to check if this helps anyway.

**Comment by hansgylling** [comment]
I have figured it out. You use the wrong syntax in your config file. I wrote some simple Go code and copied the section about add-constant from your config file and could reproduce your problem. Use this config file instead and you problem should go away:
```
[rule.add-constant]
  arguments = [{maxLitCount = "3",allowInts = "0,1,2,3,4,5"}]
```

What I wrote above about using go install has nothing to do with your bug, but could be good to know anyway.

**Comment by Integralist** [comment]
Thanks. I actually use the @latest module approach for installing. The -u version was just part of the bug issue template 😬

I'll give the updated syntax a try in my morning and will report back. Thanks for testing this out. 👍

**Comment by hansgylling** [comment]
I've already filed bug #737 to update the bug template and I will fix that soon (when I've learned a bit more about how Github works).

**Comment by Integralist** [comment]
This worked for me, thanks! 🙂
